src/runmode-dpdk.c

```diff
@@ -247,6 +259,113 @@ static void ArgumentsAddOptionAndArgument(struct Arguments *args, const char *op
     SCReturn;
 }
 
+/**
+ * Returns the first management core to be used as a main lcore
+ * @return
+ */
+static uint16_t ConfigLcoreMainGet(void)
+{
+    ThreadsAffinityType *taf = GetAffinityTypeFromName("management-cpu-set");
+    for (uint16_t i = 0; i < (uint16_t)sizeof(cpu_set_t); i++) {
+        if (CPU_ISSET(i, &taf->cpu_set)) {
+            return i;
+        }
+    }
+
+    FatalError(SC_ERR_DPDK_CONF, "No affinity set for management threads");
+}
+
+/**
+ * Convert cpu_set_t mask to an array of numbers.
+ * @param cpus - array to fill in
+ * @param sz - size of the array
+ * @return length of the new array
+ */
+static int ConfigLcoreWorkersSet(uint16_t *cpus, const size_t sz)
+{
+    int cpus_len = 0;
+    memset(cpus, 0, sz);
+    ThreadsAffinityType *taf = GetAffinityTypeFromName("worker-cpu-set");
+
+    for (uint16_t i = 0; i < (uint16_t)sizeof(cpu_set_t); i++) {
+        if (CPU_ISSET(i, &taf->cpu_set)) {
+            cpus[cpus_len++] = i;
+        }
+    }
+
+    return cpus_len;
+}
+
+/**
+ * Function converts cpu_set_t mask to a string of lcores to enable divided by a comma separator
+ * Used for EAL initialization together with "-l" parameter (format -l 2,4,6,8).
+ * @return dynamically allocated string
+ */
+static char *ConfigLcoreArgValGet(void)
+{
+    SCEnter();
+    int ret;
+    uint16_t lcore_arr[sizeof(cpu_set_t)];
+    uint16_t arr_len = 0;
+    lcore_arr[arr_len++] = ConfigLcoreMainGet();
```

Comment:
seems a bit hacky... can we just start with `lcore_arr[0] = ConfigLcoreMainGet();`
followed by `uint16_t arr_len = 1 + ConfigLcoreWorkersSet(&lcore_arr[arr_len], sizeof(cpu_set_t) - 1);` or similar?